enhancements/machine-config/machine-config-non-reconcilable-changes.md

```diff
@@ -0,0 +1,189 @@
+, that given the new configuration, will only apply changes for the fields the 
+MCD supports.
+
+### User Stories
+
+* As a cluster admin, I am adding new nodes to a long-standing cluster and I 
+would like change the partitions for the new nodes.
+* As a cluster admin, I am adding new nodes to a long-standing cluster and the 
+new hardware requires a different set of kernel arguments that I would like 
+to introduce.
+
+### Goals
+
+* Add a knob in MCO's MachineConfiguration CR to skip non-reconcilable fields 
+validation if necessary.
+
+### Non-Goals
+
+* Allow invalid Ignition/MachineConfig fields to be applied.
+* Disable non-reconcilable MCs validation by default.
+
+## Proposal
+
+Update the MachineConfiguration CR by adding a new field to the spec that 
+allows users to bypass validation for irreconcileable MachineConfig changes. 
+The field will default to the current behaviour that is to 
+validate all rendered MCs.
+
+The MachineConfig Controller and the MachingConfig MachineConfigDaemons will 
+read in runtime the new field and if the value explicitely states that the 
+validation should be skipped they will let the MC pass and get applied to 
+the nodes.
+
+MachineConfig daemons will continue to perform the already supported updates 
+to nodes, no matter if the non-reconcilable validation is skipped or not.
+Already existing nodes that receives only non-supported changes will skip the 
+update and will be considered updated.
+
+### Workflow Description
+
+When a MCO user modifies the cluster-wide MachineConfigs a new rendered 
+MachineConfig CR is created for each pool that has an association with the 
+created, modified or deleted MachineConfig. The rendered MachineConfig, before
+being applied, is validated against the Ignition Schema.
+
+After the Ignition validation is done, the non-reconcilable fields validation 
+is performed or skipped based on the proposed
+`machineConfigurationValidationPolicy` field in the MachineConfiguration CR.
+If the field is set to `Relaxed` the non-reconcilable fields validation is 
+skipped performed, otherwise is done.
+
+The non-reconcilable MC validation remains as it is with this enhancement, as 
+the [implementation](https://github.com/openshift/machine-config-operator/blob/e44d380686aee42f784a277236dbac49b083441e/pkg/controller/common/reconcile.go#L69) 
+does not change with this enhancement.
+
+### API Extensions
+
+- Update the MachineConfiguration CRD to add an enumeration field, called 
+`machineConfigurationValidationPolicy` that is used as the validation 
+skipping toggle. The field does not set a default values to let the MCO pick 
+what to do in the default case. The enumeration has only two values:
+    - Strict: Validation is always performed. This is the value the MCO will
+    use as default.
+    - Relaxed: The validation of non reconcilable fields is skipped and only 
+    the Ignition syntactic validation will be done. 
+
+### Risks and Mitigations
```

Comment:
While it is called out as a non-goal for allowing invalid Ignition/MachineConfig instances, how do you plan to prevent this? Is it possible for this proposed field to be leveraged in a what that could cause this?

Reply:
The MCO always validates all ignition and machineconfigs, and we do not plan to ever support invalid configuration. Invalid in this case means a mistake in the provided spec.